janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchSetup.java

```diff
@@ -71,6 +71,12 @@ public Connection connect(Configuration config) throws IOException {
             if (config.has(ElasticSearchIndex.MAX_RETRY_TIMEOUT)) {
                 rcb.setMaxRetryTimeoutMillis(config.get(ElasticSearchIndex.MAX_RETRY_TIMEOUT));
             }
+
+            rcb.setRequestConfigCallback(builder -> builder
+                    .setConnectTimeout(config.get(ElasticSearchIndex.MAX_CONNECT_TIMEOUT))
+                    .setSocketTimeout(config.get(ElasticSearchIndex.MAX_CONNECT_TIMEOUT))
+                    .setConnectionRequestTimeout(0));
```

Comment:
Was it necessary to set `setConnectionRequestTimeout(0)`? If so, why not set it to `ElasticSearchIndex.MAX_CONNECT_TIMEOUT`? Based off the config option description leaving this as infinite seems misleading. 

> A timeout value of zero is interpreted as an infinite timeout. A negative value is interpreted as undefined (system default).
> 
> Default: -1

Reply:
It was one of the solutions provided [in this issue](https://github.com/elastic/elasticsearch/issues/24069): https://github.com/elastic/elasticsearch/issues/24069#issuecomment-354821932

In any case I'm closing this PR since JanusGraph uses [6.6 version of ElasticSearch client in master](https://github.com/JanusGraph/janusgraph/blob/master/pom.xml#L93) where this issue [was already fixed](https://github.com/elastic/elasticsearch/pull/30384/files): https://github.com/elastic/elasticsearch/blame/6.6/client/rest/src/main/java/org/elasticsearch/client/RestClientBuilder.java#L221